vendor/gopkg.in/yaml.v2/yaml.go

```diff
@@ -13,6 +13,29 @@ import (
 	"reflect"
 	"strings"
 	"sync"
+	"sync/atomic"
+)
+
+var disabledFeatures = int32(0)
+
+// DisableFeature disables the specified feature for all callers of Unmarshal or Decode.
+func DisableFeature(feature Feature) {
+	disabled := atomic.LoadInt32(&disabledFeatures)
+	disabled = disabled | int32(feature)
+	atomic.StoreInt32(&disabledFeatures, disabled)
+}
+
+// FeatureDisabled returns true if the specified feature is disabled.
+func FeatureDisabled(feature Feature) bool {
+	return atomic.LoadInt32(&disabledFeatures)&int32(feature) == int32(feature)
+}
+
+// Feature represents a specific feature that can be passed to DisableFeature.
+type Feature int32
+
+const (
+	// AliasNodes controls use of the `*<anchor-name>` feature
+	AliasNodes Feature = 1
```

Comment:
Worth making this `1 << iota` to protect against the third Feature choosing `3`?

Reply:
done